src/main/kotlin/org/misarch/catalog/persistance/model/ProductEntity.kt

```diff
@@ -0,0 +1,46 @@
+package org.misarch.catalog.persistance.model
+
+import org.misarch.catalog.graphql.model.Product
+import org.springframework.data.annotation.Id
+import org.springframework.data.relational.core.mapping.Table
+import java.util.*
+
+/**
+ * Entity for products
+ *
+ * @property internalName internal name of the product
+ * @property isPubliclyVisible whether the product is publicly visible
+ * @property defaultVariantId id of the default variant of the product
+ * @property id unique identifier of the product
+ */
+@Table
+class ProductEntity(
+    var internalName: String,
+    var isPubliclyVisible: Boolean,
+    var defaultVariantId: UUID?,
+    @Id
+    val id: UUID?
+) {
+
+    companion object {
+        /**
+         * Querydsl entity
+         */
+        val ENTITY = QProductEntity.productEntity!!
```

Comment:
Okay, I think somewhere code is generated.
I simply have no idea where, which is not a good sign.

Reply:
is is caused by the querydsl dependency

Reply:
tbh I don't really know how to improve this.
I will mention it in the dev docs, when these exist.
But I don't know how to make it clearer here.
Do you have any ideas?

Reply:
No, unfortunately I don't have either.
For what exactly are they generated?

Reply:
basically a DSL to write SQL queries